src/Host.cpp

```diff
@@ -764,33 +767,57 @@ bool Host::installPackage(const QString& fileName, int module )
         mpEditorDialog->doCleanReset();
     }
     QFile file2;
-    if( fileName.endsWith(".zip") || fileName.endsWith(".mpackage") )
+    if(  fileName.endsWith( QStringLiteral( ".zip" ), Qt::CaseInsensitive )
+      || fileName.endsWith( QStringLiteral( ".mpackage"), Qt::CaseInsensitive ) )
     {
-        QString _home = QDir::homePath();
-        _home.append( "/.config/mudlet/profiles/" );
-        _home.append( getName() );
-        QString _dest = QString( "%1/%2/").arg( _home ).arg( packageName );
-        QDir _tmpDir;
-        _tmpDir.mkpath(_dest);
-
-        QUiLoader loader;
-        QFile file(":/ui/package_manager_unpack.ui");
-        file.open(QFile::ReadOnly);
-        pUnzipDialog = dynamic_cast<QDialog *>(loader.load( &file, 0 ) );
-        file.close();
-        if( ! pUnzipDialog ) return false;
-
-        pUnzipDialog->setWindowTitle( tr( "Unpacking package: %1" ).arg( fileName ) );
+        QString _home = QStringLiteral( "%1/.config/mudlet/profiles/%2" )
+                        .arg( QDir::homePath() )
+                        .arg( getName() );
+        QString _dest = QStringLiteral( "%1/%2/" )
+                        .arg( _home )
+                        .arg( packageName );
+        QDir _tmpDir( _home ); // home directory for the PROFILE
+        _tmpDir.mkpath( _dest );
+        // TODO: report failure to create destination folder for package/module in profile
+
+        QUiLoader loader( this );
+        QFile uiFile( QStringLiteral( ":/ui/package_manager_unpack.ui" ) );
+        uiFile.open(QFile::ReadOnly);
+        pUnzipDialog = dynamic_cast<QDialog *>(loader.load( &uiFile, 0 ) );
+        uiFile.close();
+        if( ! pUnzipDialog )
+        {
+            return false;
+        }
+
+        QLabel * pLabel = pUnzipDialog->findChild<QLabel*>( QStringLiteral( "label" ) );
+        if( pLabel )
+        {
+            if( module )
+            {
+                pLabel->setText( tr( "Unpacking module:\n\"%1\"\nplease wait..." ).arg( packageName ) );
+            }
+            else
+            {
+                pLabel->setText( tr( "Unpacking package:\n\"%1\"\nplease wait..." ).arg( packageName ) );
+            }
+        }
```

Comment:
This puts the package/module name clearly in the centre of the dialogue rather than in the title bar where it is likely to be elided and not readable in the previous code!